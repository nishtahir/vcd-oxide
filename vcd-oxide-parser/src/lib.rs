extern crate pest;
#[macro_use]
extern crate pest_derive;

use pest::Parser;
use pest_derive::Parser;

#[derive(Parser, Debug)]
#[grammar = "grammar/vcd.pest"]
pub struct ValueChangeDumpFile;

#[cfg(test)]
mod test {
    use super::*;
    use insta::assert_debug_snapshot;

    #[test]
    fn test_vcd_declaration_command() {
        let declatations = include_str!("../test/declaration_command.vcd.test");
        let file = ValueChangeDumpFile::parse(Rule::value_change_dump_definitions, declatations);
        assert_debug_snapshot!(file.unwrap())
    }

    
}
