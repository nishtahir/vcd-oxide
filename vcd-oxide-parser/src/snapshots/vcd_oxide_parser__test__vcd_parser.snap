---
source: vcd-oxide-parser/src/lib.rs
assertion_line: 21
expression: file.unwrap()
---
[
    Pair {
        rule: value_change_dump_definitions,
        span: Span {
            str: "$date\n\tFri Jan 27 10:13:28 2023\n$end",
            start: 0,
            end: 36,
        },
        inner: [
            Pair {
                rule: declaration_command,
                span: Span {
                    str: "$date\n\tFri Jan 27 10:13:28 2023\n$end",
                    start: 0,
                    end: 36,
                },
                inner: [
                    Pair {
                        rule: declaration_keyword,
                        span: Span {
                            str: "$date",
                            start: 0,
                            end: 5,
                        },
                        inner: [],
                    },
                    Pair {
                        rule: command,
                        span: Span {
                            str: "Fri Jan 27 10:13:28 2023",
                            start: 7,
                            end: 31,
                        },
                        inner: [],
                    },
                    Pair {
                        rule: end,
                        span: Span {
                            str: "$end",
                            start: 32,
                            end: 36,
                        },
                        inner: [],
                    },
                ],
            },
        ],
    },
]
